### ISSUE TITLE
v7.x Q re throw expressions

### ISSUE THREAD

**Issue by RexJaeschke** [dotnet/csharpstandard#66]
In Draft PR https://github.com/ECMA-TC49-TG2/csharpstandard/pull/65, I've added almost all the edits to support the addition of throw expressions. Here's what's missing:

1. I've added the new section "The throw expression operator" to the the Expressions clause, but it needs to be "plugged into" the grammar, so it has precedence. (Also, for now, I've put this new section under primary expressions, but that might not be correct.)

**Comment by KalleOlaviNiemitalo** [comment]
The interaction of <https://github.com/dotnet/csharpstandard/pull/65> and <https://github.com/dotnet/csharpstandard/pull/213> does not allow a throw expression as an operand of a ref conditional expression:

```csharp
ref int F(bool b, ref int i) => ref b ? ref i : ref throw null;
```

because "Both *expression*s shall be variables having the same type" and "A *throw_expression* has no type." I suppose this restriction is by design.

**Comment by BillWagner** [comment]
See PR #65 

Note that instead of "primary", the throw expression precedence should be the same as the null coalescing question.

**Comment by jskeet** [comment]
@KalleOlaviNiemitalo: It's not clear to me from your comment whether this is how the compiler actually behaves, or whether this is an area you think this needs a spec change. I want to close this issue when #65 is merged, so please could you file a new issue if you believe we need to do more work on this?

**Comment by KalleOlaviNiemitalo** [comment]
@jskeet the standard text matches compiler behaviour and I think the behaviour is reasonable so no action is required.

**Comment by BillWagner** [comment]
@jskeet PR #65 has been merged. I think that the questions raised here have been answered.

Can we close this as fixed in #65

**Comment by jskeet** [comment]
Yes, I think so.
